src/m365/outlook/commands/mailbox/mailbox-settings-set.ts

```diff
@@ -0,0 +1,204 @@
+    })).optional()),
+    userName: zod.alias('n', z.string().refine(name => validation.isValidUserPrincipalName(name), name => ({
+      message: `'${name}' is not a valid UPN.`
+    })).optional()),
+    dateFormat: z.string().optional(),
+    timeFormat: z.string().optional(),
+    timeZone: z.string().optional(),
+    language: z.string().optional(),
+    delegateMeetingMessageDeliveryOptions: z.enum(['sendToDelegateAndInformationToPrincipal', 'sendToDelegateAndPrincipal', 'sendToDelegateOnly']).optional(),
+    workingDays: z.string().transform((value) => value.split(',')).pipe(z.enum(['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']).array()).optional(),
+    workingHoursStartTime: z.string().optional(),
+    workingHoursEndTime: z.string().optional(),
+    workingHoursTimeZone: z.string().optional(),
+    autoReplyExternalAudience: z.enum(['none', 'all', 'contactsOnly']).optional(),
+    autoReplyExternalMessage: z.string().optional(),
+    autoReplyInternalMessage: z.string().optional(),
+    autoReplyStartDateTime: z.string().optional(),
+    autoReplyStartTimeZone: z.string().optional(),
+    autoReplyEndDateTime: z.string().optional(),
+    autoReplyEndTimeZone: z.string().optional(),
+    autoReplyStatus: z.enum(['disabled', 'scheduled', 'alwaysEnabled']).optional()
+  })
+  .strict();
+
+declare type Options = z.infer<typeof options>;
+
+interface CommandArgs {
+  options: Options;
+}
+
+class OutlookMailboxSettingsSetCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MAILBOX_SETTINGS_SET;
+  }
+
+  public get description(): string {
+    return 'Updates user mailbox settings';
+  }
+
+  public get schema(): z.ZodTypeAny | undefined {
+    return options;
+  }
+
+  public getRefinedSchema(schema: typeof options): z.ZodEffects<any> | undefined {
+    return schema
+      .refine(options => [options.workingDays, options.workingHoursStartTime, options.workingHoursEndTime, options.workingHoursTimeZone,
+        options.autoReplyStatus, options.autoReplyExternalAudience, options.autoReplyExternalMessage, options.autoReplyInternalMessage,
+        options.autoReplyStartDateTime, options.autoReplyStartTimeZone, options.autoReplyEndDateTime, options.autoReplyEndTimeZone,
+        options.timeFormat, options.timeZone, options.dateFormat, options.delegateMeetingMessageDeliveryOptions, options.language].filter(o => o !== undefined).length > 0, {
+        message: 'Specify at least one of the following options: workingDays, workingHoursStartTime, workingHoursEndTime, workingHoursTimeZone, autoReplyStatus, autoReplyExternalAudience, autoReplyExternalMessage, autoReplyInternalMessage, autoReplyStartDateTime, autoReplyStartTimeZone, autoReplyEndDateTime, autoReplyEndTimeZone, timeFormat, timeZone, dateFormat, delegateMeetingMessageDeliveryOptions, or language'
+      });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const data: any = {
```

Comment:
Let's move constructing the `data` object to a private function for readability...